packages/core/src/new-api/internal/reconciliation/futures/reconcileArtifactContractDeployment.ts

```diff
@@ -36,12 +37,15 @@ export function reconcileArtifactContractDeployment(
     return fail(future, "Libraries have been changed");
   }
 
-  if (!isEqual(future.value, executionState.value)) {
+  const resolvedValue =
+    typeof future.value === "bigint"
+      ? future.value
+      : (resolveModuleParameter(future.value, context) as bigint);
```

Comment:
Can we pull out this entire section as a function and reuse it for each.

That is so that we can add a check on the return result if it is a module parameter. We want to check that the value is a bigint and fail there.